File: Chapter_2/sorting_algorithms/insertion.py

arr = [1,23,1,42,4,2,132,9]


for i in range(1,len(arr)):
    j = i-1
    value = arr[i]
    # print(value)


    while j>=0 and arr[j]>=value :

        print("while")
        #shifting right by one
        arr[j+1] = arr[j]
        j-=1
        print(arr)

    #assigning in the found spot
    arr[j+1] = value

File: Chapter_2/rough/insertion.cpp
#include <iostream>

using namespace std;



void printArray(int arr[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        cout << arr[i] << " ";
    cout << endl;
}

int main() {

    int n[] = {1,2,3,3,4};

    for(int i=0 ; i<5;i+=1) {

       int j = i-1;
       int value = n[i];

       while (j>=0 && n[j]>=value) {
//            shifting to the right
            n[j+1] = n[j];
            j=j-1;
       }
       n[j+1] = value;
    }
    int size = sizeof(n)/sizeof(n[0]);

    printArray(n,size);
}